internal/pkg/daemon/bpfrecorder/bpfrecorder.go

```diff
@@ -104,14 +110,10 @@ func New(logger logr.Logger) *BpfRecorder {
 // Run the BpfRecorder.
 func (b *BpfRecorder) Run() error {
 	b.logger.Info(fmt.Sprintf("Setting up caches with expiry of %v", defaultCacheTimeout))
-	for _, cache := range []*ttlcache.Cache{
+	for _, cache := range []*ttlcache.Cache[string, string]{
 		b.containerIDCache, b.syscallNamesForIDCache,
 	} {
-		if err := b.SetTTL(cache, defaultCacheTimeout); err != nil {
-			return fmt.Errorf("set cache timeout: %w", err)
-		}
-		cache.SetCacheSizeLimit(maxCacheItems)
-		defer cache.Close() // nolint:gocritic // this is intentional
+		go cache.Start()
```

Comment:
don't we still need to set the cachesizelimit?

Reply:
We do above where we create the cache :)